backend/addcorpus/migrations/0014_corpus_has_python_definition.py

```diff
@@ -0,0 +1,18 @@
+# Generated by Django 4.2.10 on 2024-03-13 10:12
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('addcorpus', '0013_alter_field_visualizations_map'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='corpus',
+            name='has_python_definition',
+            field=models.BooleanField(default=False, help_text='whether the configuration of this corpus is determined by a Python module (some features are only available for Python-based corpora)'),
```

Comment:
Should this contain a data migration?
What happens when we (for some reason) need to roll back a few migrations, and then reapply this. Existing corpora that have set `has_python_definition=True` will then be `False` because of the default.

Reply:
Yes, I wasn't sure about this step.

I think in that situation, the value would be set to `False` in the migration, but set to `True` when you import Python corpora (which happens when you start the server).

Reply:
Oh yeah, that makes sense. In that case it's acceptable to leave the data migration out. Maybe provide a comment that this is done implicitly at server start.